portiaui/app/adapters/application.js

```diff
@@ -1,3 +1,400 @@
+                                        if (responses[type][id]) {
+                                            responses[type][id] = included;
+                                        }
+                                    } else {
+                                        filtered_included.push(included);
+                                    }
+                                }
+                                response.included = filtered_included;
+                            }
+
+                            for (let {type, snapshot, resolver} of updates) {
+                                const modelName = type.modelName;
+                                const id = snapshot.id;
+                                if (responses[modelName]) {
+                                    if (responses[modelName][id]) {
+                                        resolver.resolve({
+                                            data: responses[modelName][id]
+                                        });
+                                        continue;
+                                    }
+                                }
+                                resolver.reject(new DS.AdapterError([{
+                                    title: "Missing update confirmation",
+                                    detail: `${response}`
+                                }], "JSON API updates response was missing confirmation for " +
+                                    "an updated resource"));
+                            }
+                        }
+
+                        return response;
+                    }, error => {
+                        for (let i = 0; i < updates.length; i++) {
+                            const {resolver} = updates[i];
+                            const pointer = `/meta/updates/${i}/`;
+                            const modelError = new error.constructor(
+                                filter_update_errors(error.errors, pointer),
+                                error.message);
+                            resolver.reject(modelError);
+                        }
+                    });
+                } else {
+                    // a sub-request of the bulk operation, return a promise that
+                    // will be resolved when the main request completes.
+                    const resolver = Ember.RSVP.defer();
+                    updates.push({
+                        type,
+                        snapshot,
+                        data,
+                        resolver
+                    });
+                    return resolver.promise;
+                }
+            } else {
+                promise = this.ajax(url, method, request);
+            }
+
+            // handle our custom json api extension for listing records that were
+            // deleted in the backend and should be unloaded.
+            promise.then(response => {
+                if (response && response.links && response.links.profile &&
```

Comment:
Same logic as the UPDATES_EXTENSION above just with a different constant.

Does it make sense to create a separate function for this?